src/main/java/com/example/Library/controller/AuthenticationRestController.java

```diff
@@ -26,22 +26,22 @@
 @RequestMapping("/api/v1/auth")
 public class AuthenticationRestController {
 
-    public AuthenticationRestController(AuthenticationManager authenticationManager, UserRepository userRepository, JwtTokenProvider jwtTokenProvider) {
+    public AuthenticationRestController(AuthenticationManager authenticationManager, UserService userService, JwtTokenProvider jwtTokenProvider) {
         this.authenticationManager = authenticationManager;
-        this.userRepository = userRepository;
+        this.userService = userService;
         this.jwtTokenProvider = jwtTokenProvider;
     }
 
     private final AuthenticationManager authenticationManager;
```

Comment:
AuthenticationRestController should have only 1 dependency on AuthenticationService, all other dependencies could be moved to AuthenticationService